frontend/javascripts/test/shaders/shader_syntax.spec.ts

```diff
@@ -1,277 +1,312 @@
-import getMainFragmentShader from "viewer/shaders/main_data_shaders.glsl";
+import getMainFragmentShader, {
+  getMainVertexShader,
+  type Params,
+} from "viewer/shaders/main_data_shaders.glsl";
 import mags from "test/fixtures/mags";
 import { describe, it, beforeEach, expect } from "vitest";
 import { parser } from "@shaderfrog/glsl-parser";
 
+type ShaderFunction = (params: Params) => string;
+
 describe("Shader syntax", () => {
   const originalWarn = console.warn;
 
   interface TestContext {
-    warningEmittedCount: number;
+    warnings: { emittedCount: number };
   }
 
-  beforeEach<TestContext>(async (context) => {
+  beforeEach<TestContext>(async ({ warnings }) => {
```

Comment:
Due to the way the `it.extend` method and fixtures work in vitest, this argument had to be destructured which is why I had to introduce the inner object to be able to increase the number of emitted warnings in the context object.